From 3666cbbe30e4d4e25401a5902ae359bc2c24248b Mon Sep 17 00:00:00 2001
From: Miroslaw Stein <miroslaw.stein@isode.com>
Date: Fri, 18 Jan 2019 14:30:24 +0000
Subject: Avoid deprecated boost endianess include

As of Boost 1.69.0, boost/detail/endian.h is deprecated in favour of
boost/predef/other/endian.h, and BOOST_(LITTLE|BIG)_ENDIAN by
BOOST_ENDIAN_(LITTLE|BIG)_BYTE.

Test-Information:

Unit tests pass on Debian 9

Change-Id: If7076c559a4e35219ff97603f50b80cfbe05b29b

diff --git a/Swiften/Base/Platform.h b/Swiften/Base/Platform.h
index 4deba2b..22dff30 100644
--- a/Swiften/Base/Platform.h
+++ b/Swiften/Base/Platform.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2010 Isode Limited.
+ * Copyright (c) 2010-2019 Isode Limited.
  * All rights reserved.
  * See the COPYING file for more information.
  */
@@ -43,9 +43,9 @@
 #endif
 
 // Endianness
-#include <boost/detail/endian.hpp>
-#if defined(BOOST_LITTLE_ENDIAN)
+#include <boost/predef/other/endian.h>
+#if defined(BOOST_ENDIAN_LITTLE_BYTE)
 #define SWIFTEN_LITTLE_ENDIAN
-#elif defined(BOOST_BIG_ENDIAN)
+#elif defined(BOOST_ENDIAN_BIG_BYTE)
 #define SWIFTEN_BIG_ENDIAN
 #endif
-- 
cgit v0.10.2-6-g49f6

